test/components/views/rooms/RoomList-test.js

```diff
@@ -0,0 +1,293 @@
+import React from 'react';
+import ReactTestUtils from 'react-addons-test-utils';
+import ReactDOM from 'react-dom';
+import expect from 'expect';
+import lolex from 'lolex';
+
+import * as TestUtils from 'test-utils';
+
+import sdk from '../../../../src/index';
+import MatrixClientPeg from '../../../../src/MatrixClientPeg';
+import { DragDropContext } from 'react-beautiful-dnd';
+
+import dis from '../../../../src/dispatcher';
+import DMRoomMap from '../../../../src/utils/DMRoomMap.js';
+import GroupStore from '../../../../src/stores/GroupStore.js';
+
+import { Room, RoomMember } from 'matrix-js-sdk';
+
+function generateRoomId() {
+    return '!' + Math.random().toString().slice(2, 10) + ':domain';
```

Comment:
Probably a bit moot in this case, but in general for unit tests where nothing is stored between runs, I would probably use the same data each time to make them completely deterministic.

Reply:
IRL we discussed and decided that it's ok to do this for room IDs (as here we simulate a homeserver randomly generating unique IDs and the app should be totally ambivalent to the content of the IDs)